app/src/main/java/com/fcaputo/parkingapp/mvp/presenter/ReservationPresenter.kt

```diff
@@ -0,0 +1,70 @@
+package com.fcaputo.parkingapp.mvp.presenter
+
+import com.fcaputo.parkingapp.entity.Reservation
+import com.fcaputo.parkingapp.mvp.contract.ReservationContract
+import com.fcaputo.parkingapp.utils.DateTimeCustomFormat
+import com.fcaputo.parkingapp.utils.atTime
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import java.time.LocalDateTime
+import java.util.Calendar
+
+class ReservationPresenter(private val model: ReservationContract.Model, private val view: ReservationContract.View) : ReservationContract.Presenter {
+    init {
+        view.onSaveButton { onSaveButton() }
+        view.onSettingsButton { onSettingsButton() }
+        view.onDateInputPressed(this::onDateInputClick)
+        view.onTimeInputPressed(this::onTimeInputClick)
+        view.showParkingLotSize(model.getParkingLotSize())
+    }
+
+    override fun onDateInputClick(dateInputText: CharSequence?, editTextId: Int) {
+        //If a date has already been selected, return it, otherwise return today's date
+        val dateToShow = if (!dateInputText.isNullOrEmpty()) {
+            DateTimeCustomFormat.parseDate(dateInputText.toString())
+        } else { Calendar.getInstance() }
+
+        view.showDatePicker(dateToShow, editTextId)
+    }
+
+    override fun onTimeInputClick(timeInputText: CharSequence?, editTextId: Int) {
+        //If a time has already been selected, return it, otherwise return now's time
+        val timeToShow = if (!timeInputText.isNullOrEmpty()) {
+            DateTimeCustomFormat.parseTime(timeInputText.toString())
+        } else { Calendar.getInstance() }
+
+        view.showTimePicker(timeToShow, editTextId)
+    }
```

Comment:
This is the same method, the only thing that change is wich event you call for the view, maybe you can use an enum to determine if you are using a date or a time picker, something like this:

```
enum class PickerType {
    DATE_PICKER,
    TIME_PICKER
}

override fun onDateTimeClick(inputText: CharSequence?, value: Int, picker: PickerType) {
        val itemToShow = if (!inputText.isNullOrEmpty()) {
            when(picker) {
                  PickerType.DATE_PICKER -> DateTimeCustomFormat.parseDate(dateInputText.toString())
                  PickerType.TIME_PICKER -> DateTimeCustomFormat.parseTime(timeInputText.toString())
            }
        } else { Calendar.getInstance() }

       when(picker) {
               PickerType.DATE_PICKER -> view.showDatePicker(itemToShow, value)
               PickerType.TIME_PICKER -> view.showTimePicker(itemToShow, value)
        }
    }
```

Reply:
Done. I made a similar approach.